src/daemon/daemon.cpp

```diff
@@ -1672,23 +1674,27 @@ try // clang-format on
             auto all_ipv4 = vm.get_all_ipv4(*config->ssh_key_provider);
 
             if (is_ipv4_valid(management_ip))
-                info->add_ipv4(management_ip);
+                instance_info->add_ipv4(management_ip);
             else if (all_ipv4.empty())
-                info->add_ipv4("N/A");
+                instance_info->add_ipv4("N/A");
 
             for (const auto& extra_ipv4 : all_ipv4)
                 if (extra_ipv4 != management_ip)
-                    info->add_ipv4(extra_ipv4);
+                    instance_info->add_ipv4(extra_ipv4);
 
             auto current_release =
                 mpu::run_in_ssh_session(session, "cat /etc/os-release | grep 'PRETTY_NAME' | cut -d \\\" -f2");
-            info->set_current_release(!current_release.empty() ? current_release : original_release);
+            instance_info->set_current_release(!current_release.empty() ? current_release : original_release);
         }
         return grpc::Status::OK;
     };
 
+    mp::InstanceNames instance_names;
+    for (const auto& n : request->instances_snapshots())
+        instance_names.add_instance_name(n.instance_name());
```

Comment:
This is only valid until snapshots are part of the deal. Could you please add a `TODO@snapshots`?